docs/source/daml/reference/choices.rst

```diff
@@ -72,10 +72,57 @@ Controllers
 
   The conjunction of **all** the parties are required to authorize when this choice is exercised.
 
-.. _daml-ref-anytime:
+.. _daml-ref-consumability:
+
+Consumability
+=============
+
+If no qualifier is present, choices are *consuming*: the contract is archived before the evaluation of the choice body.
+
+Note that the contracts created in the choice body are divulged to the signatories and controllers of the contracts as well as to the observers of the contract on which the choice was exercised. This behavior differs from that of *preconsuming* and *postconsuming* choices.
+
+Preconsuming choices
```

Comment:
Maybe it would make sense to mention explicitly that pre- and post-consuming choices are just syntactic sugar for non-consuming choices + `archive self`?

Reply:
That would make sense. I would not use the term "syntactic sugar" though but rather something like "you can think of it as if ..."

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4134/commits/ab6062e4e6c8dce85565d5b573495f1ed95d7e9f

Let me know if you find the wording suitable.